mParticle-Google-Analytics-Firebase/MPKitFirebaseAnalytics.m

```diff
@@ -0,0 +1,340 @@
+#import "MPKitFirebaseAnalytics.h"
+#import "Firebase.h"
+#import "MPEnums.h"
+
+@interface MPKitFirebaseAnalytics()
+
+@property (nonatomic, strong, readwrite) FIROptions *firebaseOptions;
+
+@end
+
+@implementation MPKitFirebaseAnalytics
+
+static NSString *const kMPFIRUserIdValueCustomerID = @"customerId";
+static NSString *const kMPFIRUserIdValueEmail = @"email";
+static NSString *const kMPFIRUserIdValueMPID = @"mpid";
+static NSString *const kMPFIRUserIdValueDeviceStamp = @"deviceApplicationStamp";
+
+#pragma mark Static Methods
+
++ (NSNumber *)kitCode {
+    return @(MPKitInstanceFirebaseAnalytics);
+}
+
++ (void)load {
+    MPKitRegister *kitRegister = [[MPKitRegister alloc] initWithName:@"FirebaseAnalytics" className:@"MPKitFirebaseAnalytics"];
+    [MParticle registerExtension:kitRegister];
+}
+
+
+- (MPKitExecStatus *)execStatus:(MPKitReturnCode)returnCode {
+    return [[MPKitExecStatus alloc] initWithSDKCode:self.class.kitCode returnCode:returnCode];
+}
+
+#pragma mark MPKitInstanceProtocol methods
+- (MPKitExecStatus *)didFinishLaunchingWithConfiguration:(NSDictionary *)configuration {
+    _configuration = configuration;
+    
+    if ([FIRApp defaultApp] == nil) {
+        static dispatch_once_t FirebasePredicate;
+        
+        dispatch_once(&FirebasePredicate, ^{
+            NSString *googleAppId = configuration[kMPFIRGoogleAppIDKey];
+            NSString *gcmSenderId = configuration[kMPFIRSenderIDKey];
+            
+            if (googleAppId && ![googleAppId isEqualToString:@""]) {
+                if (!gcmSenderId || [gcmSenderId isEqualToString:@""]) {
+                    NSLog(@"Invalid Google Project Number, %@. Using Default", googleAppId);
```

Comment:
I would just return here rather than attempting to initialize with a known-bad gcmGenderId. Also - the error message need to be updated.